### ISSUE TITLE
Write Errors

### ISSUE THREAD

**Issue by nhproject** [influxdata/influxdb#5876]
Hi,

We work with the TICK system - Telegraf 0.10, InfluxDB 0.10 & Kapacitor 0.10.
Specifically, we have a cluster of three InfluxDB machines.

Last week, we started to get write errors on one of them, and its log says:
``` "...write failed for shard 163 on node 1: queue is full...failed to store statistics: write failed: queue is full..." ```
We receive 6 write errors per minute in this machine.

Today, another machine started to post write errors, and its log says:
``` "...write failed for shard 162 on node 5: timed out waiting for free connection...failed to store statistics: write failed: timed out waiting for free connection..." ```
We receive 6 write errors per minute in this machine.

I'll also mention that the memory in those machines is twice than in the third machine (which has no errors, for now).

I'd like your help about that issue.

Thanks,

Nitz

**Comment by rossmcdonald** [comment]
@nhproject Thanks for letting us know. Can you create a gist with the logs from one of the servers experiencing the write issue?

**Comment by rossmcdonald** [comment]
It looks like this is a hinted-handoff issue, where the HH queue is filling up without being emptied. The specific error seems to be [here](https://github.com/influxdata/influxdb/blob/master/services/hh/queue.go#L18).

@nhproject Can you provide the output from a `SHOW SERVERS` command?

**Comment by panda87** [comment]
```
> show servers
name: data_nodes
----------------
id	http_addr		tcp_addr
1	172.21.0.57:8086	172.21.0.57:8088
3	172.21.0.58:8086	172.21.0.58:8088
5	172.21.1.140:8086	172.21.1.140:8088


name: meta_nodes
----------------
id	http_addr		tcp_addr
1	172.21.0.57:8091	172.21.0.57:8088
2	172.21.0.58:8091	172.21.0.58:8088
4	172.21.1.140:8091	172.21.1.140:8088
```

**Comment by panda87** [comment]
We are working together

**Comment by nhproject** [comment]
<img width="1059" alt="screen shot 2016-03-03 at 11 27 40" src="https://cloud.githubusercontent.com/assets/15214994/13490167/a1787dd0-e133-11e5-9067-6ef4ae12a61e.png">


<img width="1071" alt="screen shot 2016-03-03 at 11 28 22" src="https://cloud.githubusercontent.com/assets/15214994/13490172/a9771e10-e133-11e5-8aa5-1c048e2d74e0.png">


@corylanou @panda87 I think that's the cause for the errors. As Cory mentioned, the HH queue is probably filling up without being emptied.

How can we solve it? what should we do?

**Comment by nhproject** [comment]
@rossmcdonald Hi, We are losing data because of this error, The interval is 10s and it seems that Influx drops some of the data..Do you have any update on this issue? How can we solve the data loss?

**Comment by rossmcdonald** [comment]
@nhproject You can extend the cluster `write-timeout` to a higher threshold (maybe `15s`) to avoid dropped writes. I'm not sure what would be causing a HH backup, though. The error:

```
timed out waiting for free connection
```

Sounds like connections aren't being closed properly, though I've never heard of this happening before with HH. Is the HH queue dropping at all, or is it continuing to grow?

And this occurring every 10s is very peculiar. What kind of data are you receiving? If it's Telegraf data, how many Telegraf clients?

**Comment by nhproject** [comment]
> You can extend the cluster write-timeout to a higher threshold (maybe 15s) to avoid dropped writes.

I will try it, thanks!

> Sounds like connections aren't being closed properly, though I've never heard of this happening before with HH.

1. Do we need to increase the max size of the HH?
2. Will version 0.10.3 solve this connections issue? Or should we wait for a further version (0.11)?

> Is the HH queue dropping at all, or is it continuing to grow?

I'm attaching some useful graphs that are related:

![screen shot 2016-03-10 at 9 47 09 am](https://cloud.githubusercontent.com/assets/15214994/13663402/09cefdf2-e6aa-11e5-8a6b-aafa2325d1e9.png)

![screen shot 2016-03-10 at 9 47 31 am](https://cloud.githubusercontent.com/assets/15214994/13663410/17a73ffc-e6aa-11e5-98a3-dde7284fcd5c.png)

The time window here is "last 12h",
The aggregation is by 1m.

Why are we getting those weird spikes every X hours? 

> What kind of data are you receiving? If it's Telegraf data, how many Telegraf clients?

We are using Telegraf 0.10.x. Some metrics are being sampled every 10 sec (System...), some are being sampled every 1 min (MySQL...). We have ~50 Telegraf clients. We send the metrics from Telegraf to Kafka and then consume the metrics from the telegraf topic in Kafka into InfluxDB.

One more interesting issue - We are getting these errors EVERY DAY ON THE SAME HOURS:

![screen shot 2016-03-10 at 9 56 17 am](https://cloud.githubusercontent.com/assets/15214994/13663656/6dd9904a-e6ab-11e5-8017-a25ce1bd5637.png)

![screen shot 2016-03-10 at 9 57 02 am](https://cloud.githubusercontent.com/assets/15214994/13663666/78c892ee-e6ab-11e5-8598-5ca63f774c74.png)

In the above snapshots you can see time window of now() - 3d
You can see the spikes in the same hours, more or less.
What's the reason for this pattern?

Thanks for all your help!

Nitz

**Comment by PierreF** [comment]
I have the same issue:  errors "2016/03/01 16:31:33 write failed for shard 162 on node 5: timed out waiting for free connection" and hinted-handoff which keep growing (or never reduce even if I stop all inputs).

I was able to reproduce with the following pattern:

* start all nodes fresh, if there had HH backlog it will be processed
* start influx_stress on one node. keep it running for about 30 seconds, then stop it
* kill/stop another node
* restart influx_stress on the first node, error show appear
* restart the stopped node : HH never reduce / still grow is influx_stress is still running.

I've made a fix in #6361

**Comment by jsternberg** [comment]
Clustering [has been removed](https://influxdata.com/blog/update-on-influxdb-clustering-high-availability-and-monetization/) from the open source version so I'm going to close this issue. If you need an open source solution for high availability, please try the [relay](http://docs.influxdata.com/influxdb/latest/high_availability/relay/). Thanks.
